src/components/VmDialog/index.js

```diff
@@ -201,8 +203,15 @@ class VmDialog extends React.Component {
   }
 
   onChangeVmName (event) {
-    this.setState({ name: event.target.value, isChanged: true })
+    const nameCandidate = event.target.value
+
+    if (isVmNameValid(nameCandidate)) {
+      this.setState({ name: event.target.value, isChanged: true, vmNameErrorText: null })
```

Comment:
event.target.value --> nameCandidate

Or we can even set just the temporary value for vmNameErrorText and call setState() just once

Reply:
done, done